### ISSUE TITLE
games-strategy/spring -> Gentoo Portage

### ISSUE THREAD

**Issue by specing** [springlobby/overlay#25]
Any volunteers for proxy maintainership with me? I would like to upstream ebuilds for all the spring versions currently in use on the official lobby (104, 103, 101 (PD), 100 (PA,PD), 98 (XTA)) and the dev one (9999). Hopefully Jools updates XTA before all the issues below are resolved.
There will be a separate issue for springlobby -> Portage.

There is also the question of having spring ebuilds in the first place when lobbies can automatically download them on demand. Nevertheless I want to disable that auto-downloading until it can be done securely, despite Gentoo leadership not caring about security.

reading: 

- https://wiki.gentoo.org/wiki/Project:Proxy_Maintainers/User_Guide
- https://wiki.gentoo.org/wiki/Gentoo_GitHub

Current TODO:
repoman -dx full:
```

RepoMan scours the neighborhood...
  IUSE.invalid [fatal]          10
   games-strategy/spring/spring-100.0.ebuild: threaded
   games-strategy/spring/spring-100.0.ebuild: lto
   games-strategy/spring/spring-101.0.ebuild: threaded
   games-strategy/spring/spring-101.0.ebuild: lto
   games-strategy/spring/spring-103.0.ebuild: threaded
   games-strategy/spring/spring-103.0.ebuild: lto
   games-strategy/spring/spring-104.0.ebuild: threaded
   games-strategy/spring/spring-104.0.ebuild: lto
   games-strategy/spring/spring-9999.ebuild: threaded
   games-strategy/spring/spring-9999.ebuild: lto
  RESTRICT.invalid              5
   games-strategy/spring/spring-100.0.ebuild: nomirror
   games-strategy/spring/spring-101.0.ebuild: nomirror
   games-strategy/spring/spring-103.0.ebuild: nomirror
   games-strategy/spring/spring-104.0.ebuild: nomirror
   games-strategy/spring/spring-9999.ebuild: nomirror
  dependency.missingslot        10
   games-strategy/spring/spring-100.0.ebuild: RDEPEND: 'virtual/jdk' matches more than one slot, please specify an explicit slot and/or use the := or :* slot operator
   games-strategy/spring/spring-100.0.ebuild: RDEPEND: '>=media-libs/glew-1.6' matches more than one slot, please specify an explicit slot and/or use the := or :* slot operator
   games-strategy/spring/spring-101.0.ebuild: RDEPEND: 'virtual/jdk' matches more than one slot, please specify an explicit slot and/or use the := or :* slot operator
   games-strategy/spring/spring-101.0.ebuild: RDEPEND: '>=media-libs/glew-1.6' matches more than one slot, please specify an explicit slot and/or use the := or :* slot operator
   games-strategy/spring/spring-103.0.ebuild: RDEPEND: 'virtual/jdk' matches more than one slot, please specify an explicit slot and/or use the := or :* slot operator
   games-strategy/spring/spring-103.0.ebuild: RDEPEND: '>=media-libs/glew-1.6' matches more than one slot, please specify an explicit slot and/or use the := or :* slot operator
   games-strategy/spring/spring-104.0.ebuild: RDEPEND: 'virtual/jdk' matches more than one slot, please specify an explicit slot and/or use the := or :* slot operator
   games-strategy/spring/spring-104.0.ebuild: RDEPEND: '>=media-libs/glew-1.6' matches more than one slot, please specify an explicit slot and/or use the := or :* slot operator
   games-strategy/spring/spring-9999.ebuild: RDEPEND: 'virtual/jdk' matches more than one slot, please specify an explicit slot and/or use the := or :* slot operator
   games-strategy/spring/spring-9999.ebuild: RDEPEND: '>=media-libs/glew-1.6' matches more than one slot, please specify an explicit slot and/or use the := or :* slot operator
  ebuild.minorsyn               5
   games-strategy/spring/spring-100.0.ebuild: Useless blank line on line: 61
   games-strategy/spring/spring-101.0.ebuild: Useless blank line on line: 62
   games-strategy/spring/spring-103.0.ebuild: Useless blank line on line: 62
   games-strategy/spring/spring-104.0.ebuild: Useless blank line on line: 62
   games-strategy/spring/spring-9999.ebuild: Useless blank line on line: 62
  metadata.bad [fatal]          2
   games-strategy/spring/metadata.xml: DOCTYPE: SYSTEM should refer to 'http://www.gentoo.org/dtd/metadata.dtd', not 'https://www.gentoo.org/dtd/metadata.dtd'
   games-strategy/spring/metadata.xml: line: 4, Element 'herd': This element is not expected. Expected is one of ( longdescription, maintainer, slots, upstream, use ).
  metadata.warning              1
   games-strategy/spring/metadata.xml: unused local USE-description: 'multithreaded'

Note: use --without-mask to check KEYWORDS on dependencies of masked packages

Please fix these important QA issues first.
RepoMan sez: "Make your QA payment on time and you'll never see the likes of me."
```

**Comment by specing** [comment]
All issues should be resolved now, some missing libs may have to be added later on.

**Comment by abma** [comment]
no pull request? no reference where this repo was added to gentoo?!

**Comment by abma** [comment]
then this issue is not solved :-P

**Comment by specing** [comment]
Looks like OursDesCavernes is ahead of the game: https://github.com/gentoo/gentoo/pull/7558

**Comment by OursDesCavernes** [comment]
Let's co-proxy-maint this one !

**Comment by mfld-fr** [comment]
Hello,

The `ebuild` for `games-util/springlobby-0.269` sourced from Gentoo mainstream does not compile any more. Three bugs are still opened on Bugzilla : https://bugs.gentoo.org/buglist.cgi?quicksearch=springlobby. As maintaining an `ebuild` is easier in an overlay (because IMHO Gentoo reviewers are too strict for non-core & 'fun' packages like these ones), I propose to drop the mainstream `ebuild` and focus the effort on this overlay. Any comment ?

**Comment by abma** [comment]
springlobby 0.269 heavily outdated: the current version of springlobby is 0.273. very likely a lot of stuff doesn't work in 0.269 any more.

IMHO drop the ebuild for 0.269 and use 0.273. If there are still any errors, please report them upstream to https://github.com/springlobby/springlobby/issues

**Comment by mfld-fr** [comment]
@abma : submitted minimal changes to successfully build the 0.273 on latest Gentoo in PR #32

**Comment by mfld-fr** [comment]
@specing @OursDesCavernes : could you please tell me if you are still active as proxy maintainers for the mainstream `springlobby` ebuild (I see no other Spring RTS related ebuild in the Gentoo official repository) ?

**Comment by mfld-fr** [comment]
@abma : in order to register this overlay in the Gentoo database (https://github.com/gentoo/api-gentoo-org/blob/master/files/overlays/repositories.xml), could you please tell me what are the Spring RTS 'officials' for the following `XXXXX` fields ?
```
<repo quality="experimental" status="unofficial">
      <name>spring>/name>
      <description lang="en">Spring RTS Gentoo overlay</description>
      <homepage>https://github.com/springlobby/overlay</homepage>
      <owner type="person">
        <email>XXXXX</email>
        <name>XXXXX</name>
      </owner>
      <source type="git">https://github.com/springlobby/overlay.git</source>
      <source type="git">git+ssh://git@github.com/springlobby/overlay.git</source>
      <feed>https://github.com/springlobby/overlay/commits/master.atom</feed>
    </repo>
```

**Comment by abma** [comment]
hm, i fear thats me: is there an alternative of providing an email address like a link to this issue tracker?

**Comment by mfld-fr** [comment]
@abma : yeah, understand the concern... mmm... any generic email address for the Spring RTS project, not a personal one ?

**Comment by mfld-fr** [comment]
@abma : yeah, understand the concern... but overlay owner name and email are required by Gentoo... hmmm... any generic email address for the Spring RTS project, not a personal one ? I am thinking about a kind of 'gentoo@springrts.com' that would be redirected to the currently active guy to maintain the overlay and Gentoo related stuff ?

**Comment by abma** [comment]
springrts.com isn't configured for receiving mails, so please use spring@abma.de.

thank you!

**Comment by mfld-fr** [comment]
Thank you @abma ! You can now remove your email address from this issue to avoid spam, as I already requested your account on Gentoo Bugzilla (please confirm in the email from bugs.gentoo.org that you should have received) and submitted the registration of this overlay in https://github.com/gentoo/api-gentoo-org/pull/455.

**Comment by mfld-fr** [comment]
Mainstream removal requested in https://github.com/gentoo/gentoo/pull/23483 in favor of this overlay.

**Comment by mfld-fr** [comment]
@abma : I think 2 things remains to do before closing this one through #35 :
1- fix the 105.0 source tarball on the DL server to unblock `spring-105.0.ebuild`
2- merge the PR spring/spring#568 for a cleaner `spring-9999.ebuild`
Pausing my 'spring' activity until then...
